### PR TITLE
everything except the phases

### PR DESCRIPTION
**Pull request by brine**
I removed all the commits that involved the phase/turn changes, this stuff should be safe to merge and has been tested in the areas that they affect.

### CODE REVIEW

**Comment by brine** [comment]
this is ready for review

**Comment by brine** [comment]
Please review this file:
https://github.com/brine/OCTGN/blob/411ad290c38da529d94d662e6c51e1aaff61cfbd/octgnFX/Octgn/Controls/RichTextBlock.cs

**Comment by kellyelton** [comment]
accepted on accident

File: octgnFX/Octgn/Scripting/Versions/3.1.0.2.py [hunk 1]
```diff
@@ -133,6 +133,17 @@ def askCard(properties = {}, operator = None, title = "Choose card"):
 	if apiResult == None: return (None, 0)
 	return (apiResult.Item1, apiResult.Item2)
 
+def queryCard(properties = {}, exact = False):
```
Comment (kellyelton) [inline octgnFX/Octgn/Scripting/Versions/3.1.0.2.py]: What does the 'exact' parameter do?

File: octgnFX/Octgn/Scripting/Versions/Script_3_1_0_2.cs [hunk 2]
```diff
@@ -962,6 +962,54 @@ public SelectMultiCardsDlg SelectMultiCard(List<int> idList, List<int> idList2,
                                     dlg.Quantity);
             });
         }
+
+        public List<string> QueryCard(Dictionary<string, List<string>> properties, bool match)
```
Comment (kellyelton) [inline octgnFX/Octgn/Scripting/Versions/Script_3_1_0_2.cs]: What does the 'match' parameter mean?

File: octgnFX/Octgn/Scripting/Versions/Script_3_1_0_2.cs [hunk 3]
```diff
@@ -962,6 +962,54 @@ public SelectMultiCardsDlg SelectMultiCard(List<int> idList, List<int> idList2,
...
+        {
+            return QueueAction(() =>
+            {
+                var Cards = new List<string>();
+
+                if (match == false) // OR operator
+                {
+                    foreach (var p in properties)
+                    {
+                        if (p.Key.ToLower() == "model")
```
Comment (kellyelton) [inline octgnFX/Octgn/Scripting/Versions/Script_3_1_0_2.cs]: what is "model"?

File: octgnFX/Octgn.DataNew/GameSerializer.cs [hunk 4]
```diff
@@ -1153,16 +1168,6 @@ public object Deserialize(string fileName)
                 foreach (var c in doc.Document.Descendants("card"))
                 {
 					var card = new Card(new Guid(c.Attribute("id").Value), ret.Id, c.Attribute("name").Value,c.Attribute("id").Value, "",game.CardSizes["Default"],new Dictionary<string, CardPropertySet>());
-                    //var card = new Card
-                    //               {
```
Comment (kellyelton) [inline octgnFX/Octgn.DataNew/GameSerializer.cs]: thanks, we don't need any commented out code.

File: octgnFX/Octgn.DataNew/GameSerializer.cs [hunk 5]
```diff
@@ -1230,17 +1242,25 @@ public object Deserialize(string fileName)
                         foreach (var p in a.Descendants("property"))
                         {
                             var pd = game.CustomProperties.First(x => x.Name.Equals(p.Attribute("name").Value, StringComparison.InvariantCultureIgnoreCase));
-                            var newprop = pd.Clone() as PropertyDef;
-                            var val = p.Attribute("value").Value;
-                            propset.Properties.Add(newprop, val);
+                            var newpd = pd.Clone() as PropertyDef;
+
+
+
```
Comment (kellyelton) [inline octgnFX/Octgn.DataNew/GameSerializer.cs]: white space for days

**Pull request merged**